src/release/upgrade.md

```diff
@@ -19,22 +19,67 @@ $ flutter upgrade
 This command gets the most recent version of the Flutter SDK
 that's available on your current Flutter channel.
 
-If you want an even more recent version of the Flutter SDK,
-switch to a less stable Flutter channel
+If you are using the **stable** channel
+and want an even more recent version of the Flutter SDK,
+switch to the **beta** channel using `flutter channel beta`,
 and then run `flutter upgrade`.
 
+### Keeping informed
+
+We publish [migration guides] for known breaking changes.
+
+We send announcements regarding these changes to the
+[Flutter announcements mailing list][flutter-announce].
+
+To avoid being broken by future versions of Flutter,
+consider submitting your tests to our [test registry].
+
 
 ## Switching Flutter channels
 
-Flutter has three [release channels][]:
-**stable**, **beta** and **master**.
+Flutter has two release channels:
+**stable** and **beta**.
 
-{{site.alert.info}}
-  The `dev` channel was retired as of Flutter 2.8.
-{{site.alert.end}}
+### The **stable** channel
+
+We recommend the **stable** channel
+for new users
+and for production app releases.
+It is updated roughly every three months,
+and may additionally receive occasional hot fixes
+for high-severity or high-impact issues.
+
+The Flutter team's plugins and packages are continually tested against the latest **stable** release.
+
+The latest documentation for the **stable** branch is at: <https://api.flutter.dev>
+
+### The **beta** channel
+
+The **beta** channel has the latest stable release.
+This is the most recent version of Flutter that we have heavily tested.
+This channel has passed all our public testing,
+has been verified against test suites for Google products that use Flutter,
+and has been vetted against [contributed private test suites][test registry].
+The **beta** channel receives regular hot fixes
+to address newly discovered important issues.
+
+The **beta** channel is essentially the same as the **stable** channel
+but updated monthly instead of quarterly.
+Indeed, when the **stable** channel is updated,
+it is updated to the latest **beta** release.
```

Comment:
**issues:**
- Write in active voice per [GSG](https://developers.google.com/style/voice).
- Avoid equivocation unless needed.
```suggestion
The **beta** channel has the latest stable release.
This channel includes the most recent version of Flutter
that passed the following tests:

- All our public testing.
- Test suites for Google products that use Flutter.
- All [contributed private test suites][test registry].

The **beta** channel receives regular hot fixes
to address discovered important issues.

The **beta** channel works like the **stable** channel
but updated each month instead of each quarter.
When the **stable** channel updated,
the update comes from the latest **beta** release.
```

Reply:
The problem with this way of phrasing it is that it implies that there's no other tests that it passed, and I don't want to make that implication. I'm also a bit dubious about saying a channel "includes" something (the channel is just a branch and so has at any one time just one version of Flutter). Finally my concern with saying it "works like" the stable channel is that it missed the key point, which is that it really is the same as the beta channel (literally, the commit from the beta channel goes to the stable channel). Equivocation is needed there though because there are technical subtleties that mean they're not actually identical.

I'm not sure which parts are passive voice here, could you elaborate on that?

Reply:
> I ask "who makes the update?"

Ah, I see. I use the passive voice specifically to avoid saying who or what is performing something when I think being explicit about it will just make things more complicated in ways that are not relevant to the discussion. :-)

The answer to "who makes the update" is a whole complicated topic. It's partially automated, we're moving to a two-agent system, it's specifically not _anyone_ on the team... I'm not really sure how to succinctly say who it is who's doing it! :-)

> How is the beta channel different?

I mean, to the extent that it's different, this is what this whole document is trying to explain. What we're trying to do here is convey the idea that the only relevant difference between beta and stable is the update cadence. There are, however, lots of minor differences that are tricky to explain without getting into the weeds (I tried to explain the API docs difference and the plugin testing difference by mentioning those explicitly, but there's lots of other minor differences that are more nuanced).

"The **beta** channel contains most of the same code as the **stable** channel, save features to be released in the next quarter." isn't any more accurate (I would say it's more misleading actually).

> Is it different in that includes some patches and hotfixes?

Both beta and stable branches get cherry-picks.

> does it include new features to be released in the next stable release?

We're trying to express that "beta" branch releases are stable releases, just like releases on the "stable" branch. In fact the "stable" branch release is just a copy of a previous "beta" branch release. Thus, by definition if a feature ends up in a "stable" branch release it would have previously been in a "beta" branch release.